### DISCUSSION TITLE
Questions about parallel running of egs++?

### DISCUSSION THREAD

**Discussion by CQUswl** [nrc-cnrc/EGSnrc#1157] [Questions]
Recently I want to use egs++ to write programs, but I have a concern about egs++, is to write egs++ programs can run in parallel, if so, how?

**Comment by CQUswl** [comment]
> Recently I want to use egs++ to write programs, but I have a concern about egs++, is to write egs++ programs can run in parallel, if so, how?

I wonder what the parallel command line looks like?

**Reply by rtownson** [reply]
The egs++ apps have the same parallel capabilities as all EGSnrc applications. To run in parallel on 4 cores (`-n4`) on the local machine, you could use a command like this:

```
egs-parallel --batch cpu -n4 -v --force -c 'egs_app -i slab'
```
Just replace `egs_app -i slab` with whatever command you would use to run your application and input file.

**Reply by CQUswl** [reply]
I see. Thanks a million. I'll try it when I get a chance.

**Reply by ftessier** [reply]
Issuing just `egs-parallel` with no arguments will give you a short summary of usage and available options.

**Reply by CQUswl** [reply]
> Issuing just `egs-parallel` with no arguments will give you a short summary of usage and available options.

Ok, I get it. Thank you very much!
